Code/Framework/AzToolsFramework/Tests/EditorVertexSelectionTests.cpp

```diff
@@ -300,4 +343,131 @@ namespace UnitTest
+            static_cast<AZ::u32>(AzToolsFramework::ViewportInteraction::KeyboardModifier::Ctrl));
+
+        return AzToolsFramework::ViewportInteraction::MouseInteractionEvent(
+            mouseInteraction, AzToolsFramework::ViewportInteraction::MouseEvent::Down, /*captured=*/false);
+    }
+
+    TEST_F(EditorVertexSelectionManipulatorFixture, VertexPlacedWhereIntersectionPointIsFoundWithCustomReferenceSpace)
+    {
+        const AZ::EntityId entityIdGround = CreateEntityForVertexIntersectionPlacement(*this);
+
+        // position ground
+        AzToolsFramework::SetWorldTransform(
+            entityIdGround,
+            AZ::Transform::CreateFromMatrix3x3AndTranslation(
+                AZ::Matrix3x3::CreateRotationX(AZ::DegToRad(-20.0f)) * AZ::Matrix3x3::CreateRotationY(AZ::DegToRad(-40.0f)) *
+                    AZ::Matrix3x3::CreateRotationZ(AZ::DegToRad(60.0f)),
+                AZ::Vector3(14.0f, -6.0f, 5.0f)));
+
+        // camera - 12.00, 18.00, 16.00, -38.00, -175.00
+        m_cameraState.m_viewportSize = AZ::Vector2(1280.0f, 720.0f);
+        AzFramework::SetCameraTransform(
+            m_cameraState,
+            AZ::Transform::CreateFromMatrix3x3AndTranslation(
+                AZ::Matrix3x3::CreateRotationZ(AZ::DegToRad(-175.0f)) * AZ::Matrix3x3::CreateRotationX(AZ::DegToRad(-38.0f)),
+                AZ::Vector3(12.0f, 18.0f, 16.0f)));
+
+        // create orientated and scaled transform for vertex selection entity transform
+        auto vertexSelectionTransform = AZ::Transform::CreateFromMatrix3x3AndTranslation(
+            AZ::Matrix3x3::CreateRotationZ(AZ::DegToRad(45.0f)), AZ::Vector3(14.0f, 7.0f, 5.0f));
+        vertexSelectionTransform.MultiplyByUniformScale(3.0f);
+
+        // set the initial starting position of the vertex selection
+        AzToolsFramework::SetWorldTransform(m_entityId, vertexSelectionTransform);
+
+        auto vertexScreenPositions = SetupVertices(m_entityId, *this);
+
+        // press and drag the mouse (starting where the surface manipulator is)
+        // select each vertex (by holding ctrl)
+        m_actionDispatcher->CameraState(m_cameraState)->MousePosition(vertexScreenPositions[0])->MouseLButtonDown()->MouseLButtonUp();
+
+        const auto finalPositionWorld = AZ::Vector3(14.3573294f, -8.94695091f, 7.08627319f);
+        // calculate the position in screen space of the final position of the entity
+        const auto finalPositionScreen = AzFramework::WorldToScreen(finalPositionWorld, m_cameraState);
+
+        auto middleMouseDownEvent =
+            BuildMiddleMouseDownEvent(finalPositionScreen, m_viewportManipulatorInteraction->GetViewportInteraction().GetViewportId());
+
+        // explicitly handle mouse event in vertex selection instance
+        m_vertexSelection.HandleMouse(middleMouseDownEvent);
+
+        // read back the position of the vertex now
+        AZ::Vector3 localVertex;
+        bool found = false;
```

Comment:
it doesn't hurt to also check that it `found` is true.

Reply:
Sure I can add this